csharp/ql/src/semmle/code/csharp/Preprocessor.qll

```diff
@@ -236,3 +236,95 @@ class EndRegionDirective extends PreprocessorDirective, @directive_endregion {
 
   override string getAPrimaryQlClass() { result = "EndRegionDirective" }
 }
+
+/**
+ * A branching preprocessor directive, such as `IfDirective`, `ElifDirective`, or
+ * `ElseDirective`.
+ */
+class BranchDirective extends PreprocessorDirective {
+  /** Holds if the branch is taken by the preprocessor. */
+  predicate branchTaken() { none() }
+}
+
+/**
+ * A preprocessor directive with a branching condition, such as `IfDirective` or
+ * `ElifDirective`.
+ */
+class ConditionalDirective extends BranchDirective {
```

Comment:
This is missing a base class, e.g. `ConditionalDirective_` where `private class ConditionalDirective_ = @directive_if or @directive_elif`.

Reply:
I guess this could even go to the dbscheme as well.

Reply:
Yes, now that you are making DB scheme changes anyway.